src/Illuminate/Database/Query/Builder.php

```diff
@@ -278,7 +278,7 @@ public function __construct(
     /**
      * Set the columns to be selected.
      *
-     * @param  array|mixed  $columns
+     * @param  array<mixed>|mixed  $columns
```

Comment:
This can be narrowed further to something like `string|\Illuminate\Contracts\Database\Query\Expression|self|\Illuminate\Database\Query\Builder|\Illuminate\Database\Eloquent\Builder<*>|\Illuminate\Database\Eloquent\Relations\Relation|\Closure(\Illuminate\Database\Query\Builder): mixed` but I figured that'd be way too long and very hard to read so I resorted to `mixed`. 

I think `self|QueryBuilder|EloquentBuilder<*>|Relation` can just be `BuilderContract` if the contract had all the methods that are called in [parseSub](https://github.com/laravel/framework/blob/e333cc80b8f5414daec8ef7b15c1333685c4988b/src/Illuminate/Database/Query/Builder.php#L397), so I might make a PR to make that happen at a later time.

Reply:
This is not primarily to be read by humans but by tools and for them `mixed` and your proposed alternative make a **huge** difference. If you want to make it more bite-sized, you can always resort to `@phpstan-type` or the like